Kernel/CommandLine.cpp

```diff
@@ -58,15 +58,13 @@ UNMAP_AFTER_INIT void CommandLine::add_arguments(Vector<StringView> const& args)
         if (str == ""sv) {
             continue;
         }
-
-        auto pair = str.split_view('=');
-        VERIFY(pair.size() == 2 || pair.size() == 1);
-
-        if (pair.size() == 1) {
-            m_params.set(pair[0], ""sv);
-        } else {
-            m_params.set(pair[0], pair[1]);
-        }
+        // Some boot loaders may include complex key-value pairs where the value is a composite entry,
+        // we handle this by only checking for the first equals sign in each command line parameter.
+        auto key = str.find_first_split_view('=');
```

Comment:
previously there was a `VERIFY` that you found `=`; here we would possibly fill `m_params` with garbage (empty value); not sure if we care. 😅

note that current implementation seems to cater for `KEY=` case. (which probably was your idea, right?)

Reply:
The verify asserted that `split_view` returned one or two values,
This will be the case for
* A string not containing an '=' -> the string it self
* a string with a trailing '=' -> the first apart of the string and an empty string
* Strings containing only one ''=' -> both parts if the string w/o the equal sign

What would have broken is a single '=', which is now treated as an empty key, with out a value, which dies not have any bad side effects aside from not crashing when some one leaves a ` = ` in the command line
And the multiple '=' case which is now properly handled

Reply:
(the leading '=' case shouldn be the same)